packages/cpp/src/browser/cpp-build-configurations.ts

```diff
@@ -119,6 +131,25 @@ export class CppBuildConfigurationManager {
     getConfigs(): CppBuildConfiguration[] {
         return this.cppPreferences[this.BUILD_CONFIGURATIONS_PREFERENCE_KEY] || [];
     }
+
+    /** Get the list of valid defined build configurations.  */
+    getValidConfigs(): CppBuildConfiguration[] {
+        return Array.from(this.getConfigs())
+            .filter(a => a.name !== namePlaceholder && a.directory !== directoryPlaceholder)
+            .sort((a, b) => (a.name.localeCompare(b.name)));
+    }
+
+    /** Create a new build configuration with placeholder values.  */
+    async createConfig(): Promise<void> {
+        const uri = await this.preferenceProvider.getUri();
+        if (uri) {
+            const configs = this.getConfigs();
+            const newConfigs = configs.slice(0);
+            newConfigs.push({ name: namePlaceholder, directory: directoryPlaceholder });
+            await this.preferenceProvider.setPreference(this.BUILD_CONFIGURATIONS_PREFERENCE_KEY, newConfigs);
```

Comment:
if you use statically-typed proxies for preferences, you don't need constants, ts compiler won't let you to use anything else and will give your proposals